plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/S3HiveQueryRunner.java

```diff
@@ -44,46 +44,115 @@ public static DistributedQueryRunner create(
             HostAndPort s3Endpoint,
             String s3AccessKey,
             String s3SecretKey,
+            String bucketName,
             Map<String, String> additionalHiveProperties)
             throws Exception
     {
-        return HiveQueryRunner.builder()
-                .setMetastore(distributedQueryRunner -> new BridgingHiveMetastore(
-                        new ThriftHiveMetastore(
-                                new TestingMetastoreLocator(
-                                        Optional.empty(),
-                                        hiveEndpoint),
-                                new HiveConfig(),
-                                new MetastoreConfig(),
-                                new ThriftMetastoreConfig(),
-                                new HdfsEnvironment(new HiveHdfsConfiguration(
-                                        new HdfsConfigurationInitializer(
-                                                new HdfsConfig(),
-                                                ImmutableSet.of()),
-                                        ImmutableSet.of()),
-                                        new HdfsConfig(),
-                                        new NoHdfsAuthentication()),
-                                false)))
-                .setHiveProperties(ImmutableMap.<String, String>builder()
-                        .put("hive.s3.endpoint", "http://" + s3Endpoint)
-                        .put("hive.s3.aws-access-key", s3AccessKey)
-                        .put("hive.s3.aws-secret-key", s3SecretKey)
-                        .putAll(additionalHiveProperties)
-                        .build())
-                .setInitialTables(ImmutableList.of())
+        return builder()
+                .setHiveEndpoint(hiveEndpoint)
+                .setS3Endpoint(s3Endpoint)
+                .setS3AccessKey(s3AccessKey)
+                .setS3SecretKey(s3SecretKey)
+                .setBucketName(bucketName)
+                .setHiveProperties(additionalHiveProperties)
                 .build();
     }
+
+    public static Builder builder(HiveMinioDataLake hiveMinioDataLake)
+    {
+        return builder()
+                .setHiveEndpoint(hiveMinioDataLake.getHiveHadoop().getHiveMetastoreEndpoint())
+                .setS3Endpoint(hiveMinioDataLake.getMinio().getMinioApiEndpoint())
+                .setS3AccessKey(HiveMinioDataLake.ACCESS_KEY)
+                .setS3SecretKey(HiveMinioDataLake.SECRET_KEY)
+                .setBucketName(hiveMinioDataLake.getBucketName());
+    }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static class Builder
+            extends HiveQueryRunner.Builder<Builder>
+    {
+        private HostAndPort hiveEndpoint;
+        private HostAndPort s3Endpoint;
+        private String s3AccessKey;
+        private String s3SecretKey;
+        private String bucketName;
+
+        public Builder setHiveEndpoint(HostAndPort hiveEndpoint)
```

Comment:
What is "hive endpoint"?

is it metastoreUri or something?

Reply:
renamed